master/internal/experiment.go

```diff
@@ -27,42 +27,36 @@ type (
 		create searcher.Create
 		trialSnapshot
 	}
-	trialCompletedOperation struct {
-		op      searcher.Runnable
-		metrics interface{}
+	trialReportValidation struct {
+		metrics workload.ValidationMetrics
 		trialSnapshot
 	}
-	trialValidation struct {
-		validationMetrics *workload.ValidationMetrics
-	}
-	trialCompletedWorkload struct {
-		completedMessage workload.CompletedMessage
-		completedOps     []trialCompletedOperation
-		earlyExit        *trialExitedEarly
+	trialReportEarlyExit struct {
+		reason workload.ExitedReason
 		trialSnapshot
 	}
-	// trialClosed is used to replay closes missed when the master dies between when a trial closing in
-	// its actor.PostStop and when the experiment snapshots the trial closed.
-	trialClosed struct {
+	trialReportProgress struct {
 		requestID model.RequestID
+		progress  model.PartialUnits
 	}
-	trialExitedEarly struct {
-		trialID      int
-		exitedReason workload.ExitedReason
-	}
-	trialSnapshot struct {
-		requestID model.RequestID
-		trialID   int
-		snapshot  []byte
+	trialQueryIsBestValidation struct {
```

Comment:
Would those messages be replaced by the message types that are generated by prototype?

Reply:
protobuf? yeah maybe/probably. just gonna save that for a big PR to make all this into API.